debug/src/check-props.js

```diff
@@ -0,0 +1,28 @@
+
+if (process.env.NODE_ENV !== 'production') {
+  var ReactPropTypesSecret = 'SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED',
+      loggedTypeFailures = {};
+}
+
+export function checkPropTypes(typeSpecs, values, location, componentName, getStack) {
+  if (process.env.NODE_ENV !== 'production') {
+    Object.keys(typeSpecs).forEach((typeSpecName) => {
+      let error;
+      try {
+        error = typeSpecs[typeSpecName](values, typeSpecName, componentName, location, null, ReactPropTypesSecret);
+      } catch (e) {
+        error = e;
+      }
+      if (error && !(error.message in loggedTypeFailures)) {
+        loggedTypeFailures[error.message] = true;
+        
+        let errorReport = `Failed ${location} type: ${error.message}${getStack && getStack() || ''}`;
+        if (typeof console !== 'undefined') {
```

Comment:
We can safely assume that `console` is available in both `node` and `browser` environments :100:

Reply:
Cool, simplified this in a8655d6.